### PR TITLE
PIN-3996 Validate risk analysis on EService publication

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**
It needs [PIN-4021](https://github.com/pagopa/interop-be-catalog-management/pull/228) and [PIN-3981](https://github.com/pagopa/interop-commons/pull/283) to compile

[PIN-3981]: https://pagopa.atlassian.net/browse/PIN-3981?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[PIN-4021]: https://pagopa.atlassian.net/browse/PIN-4021?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala [hunk 1]
```diff
@@ -203,6 +203,23 @@ final case class ProcessApiServiceImpl(
...
+          tenant     <- tenantManagementService.getTenantById(catalogItem.producerId)
+          tenantKind <- tenant.kind.toFuture(TenantKindNotFound(tenant.id))
+          _          <- catalogItem.riskAnalysis.traverse(risk =>
+            isRiskAnalysisFormValid(riskAnalysisForm = risk.riskAnalysisForm.toTemplate, schemaOnlyValidation = false)(
+              tenantKind.toTemplate
+            )
+          )
+        } yield ()
+      } else Future.unit
+    }
```
Comment (beetlecrunch) [inline src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala]:
```suggestion
    def verifyRiskAnalysisForPublication(catalogItem: CatalogItem): Future[Unit] = catalogItem.mode match {
      case Deliver => Future.unit
      case Receive =>
        for {
          _          <-
            if (catalogItem.riskAnalysis.isEmpty) Future.failed(EServiceRiskAnalysisIsRequired(catalogItem.id))
            else Future.unit
          tenant     <- tenantManagementService.getTenantById(catalogItem.producerId)
          tenantKind <- tenant.kind.toFuture(TenantKindNotFound(tenant.id))
          _          <- catalogItem.riskAnalysis.traverse(risk =>
            isRiskAnalysisFormValid(riskAnalysisForm = risk.riskAnalysisForm.toTemplate, schemaOnlyValidation = false)(
              tenantKind.toTemplate
            )
          )
        } yield ()
    }
```

Comment (beetlecrunch) [reply src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala]: LGTM, but I prefer to check all the cases explicitly.

**Review by beetlecrunch — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ResponseHandlers.scala [hunk 2]
```diff
@@ -189,7 +189,11 @@ object ResponseHandlers extends AkkaResponses {
       case Failure(ex: OperationForbidden.type)            => forbidden(ex, logMessage)
       case Failure(ex: EServiceNotFound)                   => notFound(ex, logMessage)
       case Failure(ex: EServiceDescriptorNotFound)         => notFound(ex, logMessage)
+      case Failure(ex: TenantNotFound)                     => notFound(ex, logMessage)
+      case Failure(ex: TenantKindNotFound)                 => notFound(ex, logMessage)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ResponseHandlers.scala]:
```suggestion
```

File: src/test/scala/it/pagopa/interop/catalogprocess/CatalogProcessSpec.scala [hunk 3]
```diff
@@ -1232,10 +1232,7 @@ class CatalogProcessSpec extends SpecHelper with AnyWordSpecLike with ScalatestR
         SpecData.catalogItem.id.toString,
         SpecData.catalogDescriptor.id.toString
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/catalogprocess/CatalogProcessSpec.scala]: we can avoid this test now

**Review by galales — APPROVED** [review]

**Review by galales — APPROVED** [review]

**Pull request merged**